mkosi/__init__.py

```diff
@@ -778,11 +778,22 @@ def install_unified_kernel(state: MkosiState, roothash: Optional[str]) -> None:
                 # ukify will have signed the kernel image as well. Let's make sure we put the signed kernel
                 # image in the output directory instead of the unsigned one by reading it from the UKI.
 
-                import pefile  # type: ignore
-                pe = pefile.PE(boot_binary, fast_load=True)
-                linux = {s.Name.decode().strip("\0"): s for s in pe.sections}[".linux"]
+                # When using a tools tree, we want to use the pefile module from the tools tree instead of
+                # requiring that python-pefile is installed on the host. So we execute python as a subprocess
+                # to make sure we load pefile from the tools tree if one is used.
+
                 # TODO: Use ignore_padding=True instead of length once we can depend on a newer pefile.
-                (state.root / state.config.output_split_kernel).write_bytes(linux.get_data(length=linux.Misc_VirtualSize))
+                pefile = dedent(
+                    f"""\
+                    import pefile
+                    from pathlib import Path
+                    pe = pefile.PE("{boot_binary}", fast_load=True)
+                    linux = {{s.Name.decode().strip("\\0"): s for s in pe.sections}}[".linux"]
+                    (Path("{state.root}") / "{state.config.output_split_kernel}").write_bytes(linux.get_data(length=linux.Misc_VirtualSize))
+                    """
+                ).replace("\n", ";")
+
+                run(["python3", "-c", pefile])
```

Comment:
Maybe marginally less ugly, but I think you could not make this this a oneliner, but pass the script via stdin
```suggestion
                )

                run(["python3"], input=pefile)
```